### DISCUSSION TITLE
How can I hide bars with null data?

### DISCUSSION THREAD

**Discussion by MattBudz** [apexcharts/apexcharts.js#2573] [Q&A]
I have a grouped bar chart where not all bars have values. This is causing some groups to have big gaps between bars. I'd like to not render those bars and move the bars that have values together.

[Here is a codepen](https://codepen.io/MattBudz/pen/RwgNxGb?editors=0010) with an example where not all bars have values.

The chart currently looks like this:
<img width="1112" alt="Screen Shot 2021-08-25 at 3 45 51 PM" src="https://user-images.githubusercontent.com/46340573/130804136-0952e51b-e400-4208-ba73-de68c5f83fc9.png">

I would like it to look like this:
![Screen-Shot-2021-08-25-at-3 45 51-PMx](https://user-images.githubusercontent.com/46340573/130804176-efd5eddc-791e-470f-86c1-edf4ae8cf637.jpg)

**Comment by alekbless** [comment]
Hello @MattBudz,

Seems that the null values is causing this, have you tried filtering them out from the series.data array?

**Reply by MattBudz** [reply]
The null values have to be there so that the correct bars are under the correct categories/groups.

**Comment by DimaTrollFace** [comment]
You can try this one:
`      plotOptions: {
        bar: {
          hideZeroBarsWhenGrouped: true,
        },
      },`
